src/routing/approved.rs

```diff
@@ -2031,73 +2036,62 @@ impl Approved {
         Ok(commands)
     }
 
-    pub fn send_user_message(
-        &mut self,
-        src: SrcLocation,
-        dst: DstLocation,
-        content: Bytes,
-    ) -> Result<Vec<Command>> {
-        if !src.contains(&self.node.name()) {
+    pub fn send_user_message(&mut self, itry: Itinerary, content: Bytes) -> Result<Vec<Command>> {
+        let are_we_src =
+            matches!(itry.src, SrcLocation::Node(_)) && itry.src.name() == self.node.name();
```

Comment:
just to confirm so this means user_message shall only be sent as `SrcLocation::Node(own_name)`, right?

Reply:
No, this is an error, fixing.